Failing server test: "Expected 404 to be 200"

> Unhandled rejection: { AssertionError: expected 404 to equal 200
    at /Users/shanetomlinson/development/fxa-content-server/tests/server/helpers/routesHelpers.js:161:20
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
  message: 'expected 404 to equal 200',
  showDiff: true,
  actual: 404,
  expected: 200 } Promise {
  <rejected> { AssertionError: expected 404 to equal 200
    at /Users/shanetomlinson/development/fxa-content-server/tests/server/helpers/routesHelpers.js:161:20
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
  message: 'expected 404 to equal 200',
  showDiff: true,
  actual: 404,
  expected: 200 } }

First saw in https://travis-ci.org/mozilla/fxa-content-server/builds/399735652, then in https://travis-ci.org/mozilla/fxa-content-server/builds/399868589

blocks #6327 
blocks #6329

Applying this patch, I see some unexpected results:

```diff
diff --git a/tests/server/helpers/routesHelpers.js b/tests/server/helpers/routesHelpers.js
index 399ae9699..2ac8dd4cc 100644
--- a/tests/server/helpers/routesHelpers.js
+++ b/tests/server/helpers/routesHelpers.js
@@ -158,7 +158,7 @@ function checkUrls(origin, resources) {
               return;
             }
 
-            assert.equal(res.statusCode, 200);
+            assert.equal(res.statusCode, 200, resource.url);
 
             // If prod-like, Check all CSS and JS (except experiments.bundle.js)
             // for SRI `integrity=` attribute on the link, and that the checksum
```

> Unhandled rejection: { AssertionError: https://firefox-source-docs.mozilla.org/mobile/android/fennec/index.html: expected 404 to equal 200

--- a/tests/server/helpers/routesHelpers.js
+++ b/tests/server/helpers/routesHelpers.js
@@ -130,6 +130,21 @@ function extractUrls(body) {
   });
 }
 
+const IGNORE_URL_REGEXPS = [
+  // Do not check support.mozilla.org URLs. Issue #4712
+  // In February 2017 SUMO links started returning 404s to non-browser redirect requests
+  /support\.mozilla\.org/,
+  // skip the livereload link in the mocha tests
+  /localhost:35729/,
+  // Ignore firefox-source-docs URLs, two URLs in the Firefox Privacy Policy are wrong.
+  // See #6330. Remove this section once https://github.com/mozilla/legal-docs/pull/1135 merges.
+  /firefox-source-docs\.mozilla\.org/
+];
+
+function isUrlIgnored (url) {
+  return IGNORE_URL_REGEXPS.find(domainRegExp => domainRegExp.test(url));
+}
+
 function checkUrls(origin, resources) {
   findCssSubResources(origin, resources)
     .then((cssSubResources) => {
@@ -151,10 +166,7 @@ function checkUrls(origin, resources) {
 
         var promise = makeRequest(resource.url, requestOptions)
           .then(function (res) {
-            if (/support.mozilla.org/.test(resource.url) || /localhost:35729/.test(resource.url)) {
-              // Do not check support.mozilla.org URLs. Issue #4712
-              // In February 2017 SUMO links started returning 404s to non-browser redirect requests
-              // Also skip the livereload link in the mocha tests
+            if (isUrlIgnored(resource.url)) {
               return;
             }
 

